lib/shopping_cart.rb

```diff
@@ -0,0 +1,46 @@
+class ShoppingCart
+    attr_reader :name, :capacity, :products
+    def initialize(name, capacity)
+        @name = name
+        @capacity = capacity.to_i
+        @products = []
+        @full = false
+    end
+
+    def add_product(product)
+        @products << product
+    end
+
+    def details
+        cart_hash = {name: @name, capacity: @capacity}
+    end
+
+    def total_number_of_products
+        @products.inject(0) {|sum, product| sum += product.quantity}
+    end
+
+    def is_full?
+        if total_number_of_products > 30
+            @full = true
+        end
+        @full
```

Comment:
I wonder if the @ full variable is necessary to keep track of. since `total_number_of_products > 30`, returns a boolean, and you'll call this method to check to see if it's full or not, you could just have line 23 (without the `if`).